src/Modifiers/GravityCamera.lua

```diff
@@ -117,6 +117,15 @@ return function(PlayerModule)
 		return upVector
 	end
 
+	function cameraObject:GetUpOrientation(): (Vector3, CFrame)
+		return upVector, upCFrame
+	end
+
+	function cameraObject:SetUpOrientation(newUpVector: Vector3, newUpCFrame: CFrame)
+		upVector = newUpVector
+		upCFrame = newUpCFrame
```

Comment:
If you're setting an `upCFrame` you don't need the `upVector` as well.

You could just do:
```luau
upCFrame = newUpCFrame
upVector = newUpCFrame.YVector -- pull this from the up cframe
targetUpVector = upVector -- match this b/c we're already at the target
```

Reply:
Done, may not be at target though if switching cameras while transitioning upVector

Reply:
Not sure I follow what you mean here?

Reply:
We don't want `targetUpVector = upVector` because it might be currently transitioning during the moment the cameras are swapped and SetCFrame is called.

Ex.) If I move my character and then switch from gun camera to classic camera while the up vector is still transitioning, I still want it to continue transitioning after the camera switch (the targetUpVector is not the current upVector).

Reply:
Could have a `shouldSetTarget: boolean?` param too though, just not sure it makes sense to set target by default given that there's a separate set function for it.